src/CodegenShapeFuture.php

```diff
@@ -0,0 +1,63 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2015-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HackCodegen;
+
+/**
+ * Generate code for a shape. Please don't use this class directly; instead use
+ * the function codegenShape_FUTURE.  E.g.:
+ *
+ * codegenShape_FUTURE(vec([
+ *   new CodegenShapeMember('x', 'int'),
+ *   new CodegenShapeMember('y', 'int'),
+ * ]))
```

Comment:
all in backticks. Also probably belongs in the codegenShape_FUTURE() documentation instead

Reply:
Also, `vec([ 1,2,3 ])` construsts a PHP array, then converts it.

You want `vec[1,2,3]` instead

Reply:
I didn't see any documentation in the `CodegenFactoryTrait`, it seemed like the pattern was to define it with the class, but happy to move this around.